### PR TITLE
Document & assert startup sequence

### PR DESCRIPTION
**Pull request by matklad**
OR
* [X] I am very sure this PR could not affect performance.

### CODE REVIEW

File: src/vsr/replica.zig [hunk 1]
```diff
@@ -25,7 +25,16 @@ const tracer = @import("../tracer.zig");
...
     view_change,
+    /// Replicas start with `.recovering` status. Normally, replica immediately
+    /// transitions to a different status. The exception is a single-node cluster,
+    /// where the replica stays in `.recovering` state until it commits all entries
+    /// from its journal.
     recovering,
+    /// Replica transitions from `.recovering` to `.recovering_head` at startup
+    /// if it finds its persistent state corrupted. In this case, replica can
+    /// not participate in consensus, as it might have forgotten some of the
+    /// messages it has sent before. Instead, it waits for a SV message to get
```
Comment (sentientwaffle) [inline src/vsr/replica.zig]: sent or received

File: src/vsr/replica.zig [hunk 2]
```diff
@@ -424,6 +433,13 @@ pub fn ReplicaType(
                     self.transition_to_recovering_head();
                 }
             }
+
+            assert(
+                self.status == .recovering and self.replica_count == 1 or
```
Comment (sentientwaffle) [inline src/vsr/replica.zig]: Can you add parens around the `and` expression? Just to make the precedence obvious.

**Comment by matklad** [comment]
ptal!

**Pull request merged**